snap/snapcraft.yaml

```diff
@@ -73,46 +97,39 @@ parts:
     shell-flags: ['-eux', '-o', 'pipefail']
     build-packages: [wget]
     shell-command: |
-      export ARCH=`arch`
-
-      if [ $ARCH = "x86_64" ] ; then
-        export PACKAGE="glide-v0.12.3-linux-amd64.tar"
-      elif [ $ARCH = "armhf" ] ; then
-        export PACKAGE="glide-v0.12.3-linux-armv7.tar"
-      elif [ $ARCH = "aarch64" ] ; then
-        export PACKAGE="glide-v0.12.3-linux-arm64.tar"
-      elif [ $ARCH = "i386" ] ; then
-        export PACKAGE="glide-v0.12.3-linux-386.tar"
-      fi
-
-      pushd $SNAPCRAFT_PART_INSTALL
-      wget "https://github.com/Masterminds/glide/releases/download/v0.12.3/$PACKAGE.gz"
-      gunzip "./$PACKAGE.gz"
-      tar xvf "./$PACKAGE" --strip-components=1
-      popd
-    stage:
-     - glide
-     - -*.tar
+      case $SNAPCRAFT_ARCH_TRIPLET in
+        aarch64*)
+          export PACKAGE="glide-v0.12.3-linux-arm64.tar.gz";;
+        arm*)
+          export PACKAGE="glide-v0.12.3-linux-armv7.tar.gz";;
+        i386*|i686*)
+          export PACKAGE="glide-v0.12.3-linux-386.tar.gz";;
+        x86_64*)
+          export PACKAGE="glide-v0.12.3-linux-amd64.tar.gz";;
+        *)
+          echo "Unsupported arch $SNAPCRAFT_ARCH_TRIPLET"
+          exit 1;;
+      esac
+
+      wget "https://github.com/Masterminds/glide/releases/download/v0.12.3/$PACKAGE"
```

Comment:
what happened to the gunzip?

Reply:
`tar` can handle `tar.gz` files without needing `gunzip` (or perhaps it calls gunzip internally when it sees it has a `tar.gz` file).